### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by thulasirajkomminar**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,55 @@
+provider "aws" {
+  alias = "glue"
+}
```
Comment (svanharmelen) [inline main.tf]: Is this due to a copy and paste action, of is this really needed?

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  role   = aws_iam_role.default[0].id
+  policy = var.policy
+}
+
+resource "aws_iam_role_policy_attachment" "default" {
+  provider   = aws.glue
+  count      = var.role_arn == null
+  role       = aws_iam_role.default[0].id
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+}
```
Comment (svanharmelen) [inline main.tf:34]: Maybe use the MCAF role module instead?

File: outputs.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+output "arn" {
+  value       = aws_glue_job.default.arn
+  description = "ARN of the glue job"
```
Comment (svanharmelen) [inline outputs.tf]:
```suggestion
  description = "ARN of the Glue job"
```

File: outputs.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+}
+
+output "id" {
+  value       = aws_glue_job.default.id
+  description = "The glue job name"
```
Comment (svanharmelen) [inline outputs.tf]:
```suggestion
  description = "The Glue job name"
```

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,83 @@
+variable "name" {
+  type        = string
+  description = "The name of the glue job"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "The name of the Glue job"
```

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+variable "command_name" {
+  type        = string
+  default     = "glueetl"
+  description = "The name of the job command. Defaults to glueetl"
+}
+
+variable "connections" {
+  type        = list
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  type        = list(string)
```

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  default     = []
+  description = "The list of connections used for this job"
+}
+
+variable "default_arguments" {
+  type        = map(string)
+  default     = {}
+  description = "The map of default arguments for this job"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "A map with default arguments for this job"
```

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+variable "command_name" {
+  type        = string
+  default     = "glueetl"
+  description = "The name of the job command. Defaults to glueetl"
+}
+
+variable "connections" {
+  type        = list
+  default     = []
+  description = "The list of connections used for this job"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "A list with connections for this job"
```

File: variables.tf [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+variable "default_arguments" {
+  type        = map(string)
+  default     = {}
+  description = "The map of default arguments for this job"
+}
+
+variable "max_capacity" {
+  type        = number
+  default     = 10
+  description = "The maximum number of AWS Glue data processing units (DPUs) that can be allocated when this job runs"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "The maximum number of data processing units that can be allocated"
```

File: variables.tf [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+variable "max_retries" {
+  type        = number
+  default     = 0
+  description = "The maximum number of times to retry this job if it fails"
+}
+
+variable "policy" {
+  type        = string
+  default     = null
+  description = "A valid glue policy JSON document"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "A valid Glue policy JSON document"
```

File: variables.tf [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+  validation {
+    condition     = (var.command_name == "glueetl" && (var.max_capacity > 2 && var.max_capacity < 100)) || (var.command_name == "pythonshell" && (var.max_capacity == 0.0625 || var.max_capacity == 1))
+    error_message = "When command_name is glueetl you can allocate from 2 to 100 DPUs. When command_name is pythonshell you can allocate either 0.0625 or 1 DPU"
+  }
+}
+
+variable "max_retries" {
+  type        = number
+  default     = 0
+  description = "The maximum number of times to retry this job if it fails"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "The maximum number of times to retry a failing job"
```

File: variables.tf [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+  validation {
+    condition     = var.role_arn == null
+    error_message = "policy is required if you don't specify a role_arn"
+  }
+}
+
+variable "python_version" {
+  type        = string
+  default     = "3"
+  description = "The Python version being used to execute a Python shell job. Allowed values are 2 or 3"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "The Python version (2 or 3) being used to execute a Python shell job"
```

File: variables.tf [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+  validation {
+    condition     = var.python_version == "2" || var.python_version == "3"
+    error_message = "The python_version value must be 2 or 3"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
    error_message = "The 'python_version' value must be 2 or 3"
```

File: variables.tf [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+variable "policy" {
+  type        = string
+  default     = null
+  description = "A valid glue policy JSON document"
+
+  validation {
+    condition     = var.role_arn == null
+    error_message = "policy is required if you don't specify a role_arn"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
    error_message = "Policy is required if you don't specify a 'role_arn'"
```

File: variables.tf [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+variable "max_capacity" {
+  type        = number
+  default     = 10
+  description = "The maximum number of AWS Glue data processing units (DPUs) that can be allocated when this job runs"
+
+  validation {
+    condition     = (var.command_name == "glueetl" && (var.max_capacity > 2 && var.max_capacity < 100)) || (var.command_name == "pythonshell" && (var.max_capacity == 0.0625 || var.max_capacity == 1))
+    error_message = "When command_name is glueetl you can allocate from 2 to 100 DPUs. When command_name is pythonshell you can allocate either 0.0625 or 1 DPU"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
    error_message = "When 'command_name' is 'glueetl' you can allocate from 2 to 100 DPUs. When 'command_name' is 'pythonshell' you can allocate either 0.0625 or 1 DPU"
```

File: variables.tf [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+  validation {
+    condition     = var.python_version == "2" || var.python_version == "3"
+    error_message = "The python_version value must be 2 or 3"
+  }
+}
+
+variable "role_arn" {
+  type        = string
+  default     = null
+  description = "An optional glue execution role"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "An optional Glue execution role"
```

File: variables.tf [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+variable "script_location" {
+  type        = string
+  description = "Specifies the S3 path to a script that executes a job"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "Specifies the S3 path to the script that is executed by this job"
```

File: variables.tf [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+variable "version" {
+  type        = string
+  default     = "1.0"
+  description = "The version of glue to use"
```
Comment (svanharmelen) [inline variables.tf]:
```suggestion
  description = "The Glue version to use"
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]
A lot of cosmetic suggestions, but also 2 questions about the used TF config.

File: main.tf (hunk 1 above)
Comment (thulasirajkomminar) [reply main.tf]: do you mean the alias or the provider itself? I see in all MCAF modules a provider is used with alias

File: main.tf (hunk 2 above)
Comment (thulasirajkomminar) [reply main.tf:34]: Can i use count in the MCAF role module?

File: main.tf (hunk 1 above)
Comment (svanharmelen) [reply main.tf]: A provider (both with and without an alias) is only needed when you need to create the resources with a custom provider. If you can use the already initialized parent provider, this is almost always a better choice.

So in this case I would remove the provider block until (if ever) we need to be able to pass a specific provider.

File: main.tf (hunk 2 above)
Comment (svanharmelen) [reply main.tf:34]: Ah, I see... You only create the role if no `role_arn` is given. In that case I think this is fine. But you need to fix line 35 as it misses the `? 1 : 0` part

File: main.tf (hunk 1 above)
Comment (thulasirajkomminar) [reply main.tf]: thanks for the explanation

**Review by svanharmelen — COMMENTED** [review]
It think the only thing remaining is adding a `LICENSE` and a `version.tf` file

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**